banyan/src/store/thread_local_zstd.rs

```diff
@@ -64,10 +34,21 @@ impl DecompressionState {
     where
         F: FnMut(&[u8]) -> R,
     {
-        let len = self.decompress(compressed)?;
-        let result = f(&self.buffer[0..len]);
-        self.buffer.truncate(MIN_CAPACITY);
-        self.buffer.shrink_to_fit();
+        let capacity = Decompressor::upper_bound(compressed)
+            .unwrap_or(MAX_CAPACITY)
+            .min(MAX_CAPACITY);
+        let mut tmp = Vec::new();
+        let buffer = if capacity <= MIN_CAPACITY {
+            &mut self.buffer[..]
+        } else {
+            tmp.resize(capacity, 0u8);
+            &mut tmp
+        };
+
+        let span = tracing::debug_span!("decompress_and_transform");
```

Comment:
include the `tmp.resize` in this span?

Reply:
yes, I guess it should be included. I was even thinking of having a span "decompress" which includes buffer management and decompression, and having the caller do any spans for the transform.